src/main/java/org/antlr/intellij/plugin/parsing/PreviewParser.java

```diff
@@ -75,4 +78,23 @@ public Token matchWildcard() throws RecognitionException {
 		lastSuccessfulMatchState = getState();
 		return super.matchWildcard();
 	}
+
+	@Override
+	public void action(RuleContext _localctx, int ruleIndex, int actionIndex) {
+		warnAboutPredicatesAndActions();
+	}
+
+	@Override
+	public boolean sempred(RuleContext _localctx, int ruleIndex, int actionIndex) {
+		warnAboutPredicatesAndActions();
+		return super.sempred(_localctx, ruleIndex, actionIndex);
+	}
+
+	private void warnAboutPredicatesAndActions() {
+		if (!warnedAboutPredicates) {
+			notifyErrorListeners("WARNING: predicates and actions are not run by this interpreter. " +
```

Comment:
I suggest using correct location of action/sempred (pass it to the method), not the first line.

Reply:
It seems to me that it's already the case:

```java
	public final void notifyErrorListeners(String msg)	{
		notifyErrorListeners(getCurrentToken(), msg, null);
	}
```

I can see the correct line:column in the generated message, as you can see in https://github.com/antlr/intellij-plugin-v4/issues/523#issuecomment-1111449850

Reply:
I see but I'm not sure it's correct. It should point line:column in the grammar not in the input.

Reply:
I was actually thinking of a generic message saying that there are actions or predicates in the grammar. If there are multiple, I'm not sure a single line and call them from the grammar will be useful and could be misleading.

The mechanism @bjansen proposes is better in the sense that it warns only if an action/pred is seen during the parse, rather than generically if there are ever any actions visible.

@KvanTTT 's Point is still valid though that it's pointing into the input stream with `getCurrentToken()`. On the other hand, maybe it makes sense to simply make it more specific that upon reaching input token foo, we bypassed an action?

Reply:
> it warns only if an action/pred is seen during the parse

That was indeed what I wanted to achive.

I guess if an action is not run, there's a high probability there will be parse errors. So if we use `getCurrentToken()`, doesn't it make sense to "group" two error messages on the same token? One saying "hey we didn't expect this token here", and the other saying "but wait we skipped an action so it might be what's causing the error".

OTOH if we point to some location in the grammar, we're not actually "linking" the warning to any problem in the input.

Reply:
It's more likely that a predicate what caused problems but of course predicate usually don't make sense without other actions to set state.  I don't think we have handy access to map an action back into the grammar file/line/column.  I think it's fine to simply identify the problem and indicate that, at this point at the input, we have skipped something that could be important.